packages/teleport/src/NotificationContext.tsx

```diff
@@ -0,0 +1,30 @@
+import { createContext, useState } from 'react';
+
+// Notification Context
+type NotificationContextType = ReturnType<typeof makeNotificationContextValue>;
+
+const NotificationContext = createContext<NotificationContextType | null>(null);
+
+// Context Value
+const makeNotificationContextValue = () => {
+  const [notifications, setNotifications] = useState([]);
+
+  const removeNotification = (index: number) => {
+    setNotifications([
+      ...notifications.slice(0, index),
+      ...notifications.slice(index + 1),
```

Comment:
You can reduce these two `slice` calls into a single [splice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) call. 

(disregard this if you make the architecture change below)